src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java

```diff
@@ -890,9 +891,18 @@ && isApplicant1NotRepresented()
 
     @JsonIgnore
     public boolean isJudgementDateNotPermitted() {
-        return nonNull(getRespondToClaimAdmitPartLRspec().getWhenWillThisAmountBePaid())
-            && getRespondToClaimAdmitPartLRspec().getWhenWillThisAmountBePaid()
-            .atTime(DeadlinesCalculator.END_OF_BUSINESS_DAY).isAfter(LocalDateTime.now());
+        LocalDate whenWillThisAmountBePaid =
+            Optional.ofNullable(getRespondToClaimAdmitPartLRspec()).map(RespondToClaimAdmitPartLRspec::getWhenWillThisAmountBePaid).orElse(null);
+        LocalDate firstRepaymentDate = Optional.ofNullable(getRespondent1RepaymentPlan()).map(RepaymentPlanLRspec::getFirstRepaymentDate).orElse(null);
+        if (isNull(whenWillThisAmountBePaid) && isNull(firstRepaymentDate)) {
```

Comment:
Can we give this function a twist and refactor it to make it more readable and maintainable? Especially the 3 ifs in a row and the return true.

Reply:
I think it will make harder to understand all conditions in a single row

Reply:
Refactored the condition